src/Equinor.ProCoSys.IPO.Query/GetOutstandingIpos/GetOutstandingIposForCurrentPersonQueryHandler.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.IPO.Domain;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.Me;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Query.GetOutstandingIpos
+{
+    public class GetOutstandingIposForCurrentPersonQueryHandler : IRequestHandler<GetOutstandingIposForCurrentPersonQuery, Result<OutstandingIposResultDto>>
+    {
+        private readonly IReadOnlyContext _context;
+        private readonly ICurrentUserProvider _currentUserProvider;
+        private readonly IMeApiService _meApiService;
+        private readonly IPlantProvider _plantProvider;
+
+        public GetOutstandingIposForCurrentPersonQueryHandler(
+            IReadOnlyContext context,
+            ICurrentUserProvider currentUserProvider,
+            IMeApiService meApiService,
+            IPlantProvider plantProvider)
+        {
+            _context = context;
+            _currentUserProvider = currentUserProvider;
+            _meApiService = meApiService;
+            _plantProvider = plantProvider;
+        }
+
+        public async Task<Result<OutstandingIposResultDto>> Handle(GetOutstandingIposForCurrentPersonQuery request,
+            CancellationToken cancellationToken)
+        {
+            var currentUserOid = _currentUserProvider.GetCurrentUserOid();
+
+            var currentUsersFunctionalRoleCodes =
```

Comment:
move this "get" inside foreach-loop at line 50 ... no need if no invitations exists (which will be most of the cases!)

Reply:
Add a Unit test, veryfying that _meApiService.GetFunctionalRoleCodesAsync is never called when no invitations exists